presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -87,75 +63,139 @@ public CassandraClusteringPredicatesExtractor(List<CassandraColumnHandle> cluste
-                             * Prior to Cassandra 2.2, only the last clustering column can have a range of values
-                             * Take a look at how this is done in PreparedStatementBuilder.java
-                             */
-
-                            Object value = range.getSingleValue();
-
-                            CassandraType valueType = columnHandle.getCassandraType();
-                            columnValues.add(valueType.validateClusteringKey(value));
                         }
-                        return columnValues.build();
-                    },
-                    discreteValues -> {
-                        if (discreteValues.isWhiteList()) {
-                            return ImmutableSet.copyOf(discreteValues.getValues());
+                        if (!range.getHigh().isUpperUnbounded()) {
+                            switch (range.getHigh().getBound()) {
+                                case ABOVE:
+                                    throw new VerifyException("High Marker should never use ABOVE bound");
+                                case EXACTLY:
+                                    rangeConjuncts.add(CassandraCqlUtils.validColumnName(columnHandle.getName()) + " <= "
+                                            + CassandraCqlUtils.cqlValue(toCQLCompatibleString(range.getHigh().getValue()),
+                                            columnHandle.getCassandraType()));
+                                    break;
+                                case BELOW:
+                                    rangeConjuncts.add(CassandraCqlUtils.validColumnName(columnHandle.getName()) + " < "
+                                            + CassandraCqlUtils.cqlValue(toCQLCompatibleString(range.getHigh().getValue()),
+                                            columnHandle.getCassandraType()));
+                                    break;
+                                default:
+                                    throw new AssertionError("Unhandled bound: " + range.getHigh().getBound());
+                            }
                         }
-                        return ImmutableSet.of();
-                    },
-                    allOrNone -> ImmutableSet.of());
+                    }
+                }
 
-            if (!values.isEmpty()) {
-                clusteringColumnValues.add(values);
+                if (!singleValues.isEmpty()) {
+                    if (singleValues.size() == 1) {
+                        predicate = CassandraCqlUtils.validColumnName(columnHandle.getName()) + " = " + singleValues.get(0);
+                    }
+                    else {
+                        predicate = CassandraCqlUtils.validColumnName(columnHandle.getName()) + " IN ("
+                                + Joiner.on(",").join(singleValues) + ")";
+                    }
+                }
+                else if (!rangeConjuncts.isEmpty()) {
```

Comment:
I have added an extra if condition to handle this scenario and will not pass this column to Cassandra and will complete the predicate string here.